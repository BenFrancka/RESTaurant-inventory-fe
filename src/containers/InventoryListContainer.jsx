import React, { useState, useEffect } from 'react';
import InventoryList from '../components/inventory/InventoryList';
import Header from '../components/inventory/Header';
import { fetchInventoryList } from '../services/inventoryUtils';

const InventoryListContainer = () => {
  const [loading, setLoading] = useState(true);
  const [inventory, setInvenentory] = useState([]);

  useEffect(() => {
    fetchInventoryList('https://localhost.7890/api/v1/inventory')
      .then((inventory) => setInvenentory(inventory))
      .finally(() => setLoading(false));
  }, []);

  if (loading) {
    return (
      <img
        src="https://c.tenor.com/tEBoZu1ISJ8AAAAC/spinning-loading.gif"
        alt="loading-spinner"
      />
    );
  }
  return (
    <>
      <Header />
      <InventoryList inventory={inventory}/>
    </>
  );
};

export default InventoryListContainer;
