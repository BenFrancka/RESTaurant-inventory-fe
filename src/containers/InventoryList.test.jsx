import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import InventoryListContainer from './InventoryListContainer';
import { MemoryRouter } from 'react-router';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import list from '../fixtures/list.json';

const server = setupServer(
  rest.get('http://localhost:7890/api/v1/inventory', (req, res, ctx) => {
    return res(ctx.json(list));
  })
);

describe('AristSearch Container', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('should display a list of current inventory items', async () => {
    const { container } = render(
      <MemoryRouter>
        <InventoryListContainer />
      </MemoryRouter>
    );

    screen.getByAltText('loading-spinner');

    const ul = await screen.findByRole('list', { name: 'inventory-list' });
    expect(ul).not.toBeEmptyDOMElement();
    expect(container).toMatchSnapshot();
  });
});
