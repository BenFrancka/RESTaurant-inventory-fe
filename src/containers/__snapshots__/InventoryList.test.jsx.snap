// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AristSearch Container should display a list of current inventory items 1`] = `
<div>
  <header>
    <h1>
      RESTaurant Inventory 
    </h1>
    <h3>
      <a
        href="/add"
      >
        Add an Item
      </a>
    </h3>
    <h3>
      <a
        href="/"
      >
        Return to Inventory
      </a>
    </h3>
  </header>
  <ul
    aria-label="inventory-list"
  >
    <a
      href="/1"
    >
      <li>
        <h2>
          potatoes
        </h2>
        <p>
          Category:
           
          produce
        </p>
        <p>
          Item Price:
           
          5
        </p>
        <p>
          Total Items:
           
          5
        </p>
        <p>
          Total Price:
           
          25
        </p>
      </li>
    </a>
    <a
      href="/2"
    >
      <li>
        <h2>
          onions
        </h2>
        <p>
          Category:
           
          produce
        </p>
        <p>
          Item Price:
           
          20
        </p>
        <p>
          Total Items:
           
          1
        </p>
        <p>
          Total Price:
           
          20
        </p>
      </li>
    </a>
    <a
      href="/3"
    >
      <li>
        <h2>
          bacon
        </h2>
        <p>
          Category:
           
          protein
        </p>
        <p>
          Item Price:
           
          90
        </p>
        <p>
          Total Items:
           
          1
        </p>
        <p>
          Total Price:
           
          90
        </p>
      </li>
    </a>
  </ul>
</div>
`;

exports[`Inventory List Screen should display a list of current inventory items 1`] = `
<div>
  <header>
    <h1>
      RESTaurant Inventory 
    </h1>
    <h3>
      <a
        href="/add"
      >
        Add an Item
      </a>
    </h3>
    <h3>
      <a
        href="/"
      >
        Return to Inventory
      </a>
    </h3>
  </header>
  <ul
    aria-label="inventory-list"
  >
    <a
      href="/1"
    >
      <li>
        <article
          aria-label="inventory item"
        >
          <h2>
            potatoes
          </h2>
          <p>
            Category:
             
            produce
          </p>
          <p>
            Item Price:
             
            5
          </p>
          <p>
            Total Items:
             
            5
          </p>
          <p>
            Total Price:
             
            25
          </p>
        </article>
      </li>
    </a>
    <a
      href="/2"
    >
      <li>
        <article
          aria-label="inventory item"
        >
          <h2>
            onions
          </h2>
          <p>
            Category:
             
            produce
          </p>
          <p>
            Item Price:
             
            20
          </p>
          <p>
            Total Items:
             
            1
          </p>
          <p>
            Total Price:
             
            20
          </p>
        </article>
      </li>
    </a>
    <a
      href="/3"
    >
      <li>
        <article
          aria-label="inventory item"
        >
          <h2>
            bacon
          </h2>
          <p>
            Category:
             
            protein
          </p>
          <p>
            Item Price:
             
            90
          </p>
          <p>
            Total Items:
             
            1
          </p>
          <p>
            Total Price:
             
            90
          </p>
        </article>
      </li>
    </a>
  </ul>
</div>
`;
